test/run-test.ps1

```diff
@@ -1,42 +1,89 @@
+                Replace($dirSeparator, '-') +
+            $tagSuffix
 
-    $fullSdkTag="$($dockerRepo):$sdkTag"
-    $baseTag="$fullSdkTag".Replace("-sdk", '').Replace("-msbuild", '').Replace("-projectjson", '').Replace("-nanoserver", '')
+        $fullSdkTag="${dockerRepo}:${sdkTag}"
+        $baseTag=$fullSdkTag.
+            TrimEnd($tagSuffix).
+            TrimEnd("-msbuild").
+            TrimEnd("-projectjson").
+            TrimEnd("-sdk")
 
-    $timeStamp = Get-Date -Format FileDateTime
-    $appName="app$timeStamp"
-    $appDir="${repoRoot}\.test-assets\${appName}"
+        $timeStamp = Get-Date -Format FileDateTime
+        $appName="app$timeStamp"
+        $appDir="$repoRoot$dirSeparator.test-assets$dirSeparator$appName"
 
-    New-Item $appDir -type directory | Out-Null
+        New-Item $appDir -type directory | Out-Null
 
-    Write-Host "----- Testing $fullSdkTag -----"
-    docker run -t $optionalDockerRunArgs -v "$($appDir):c:\$appName" -v "$repoRoot\test:c:\test" --name "sdk-test-$appName" --entrypoint powershell "$fullSdkTag" c:\test\create-run-publish-app.ps1 "c:\$appName" "${sdkTag}"
-    if (-NOT $?) {
-        throw  "Testing $full_sdk_tag failed"
-    }
+        Write-Host "----- Testing $fullSdkTag -----"
+        docker run -it --rm `
+            -v "${appDir}:${containerRoot}${appName}" `
+            -v "${repoRoot}${dirSeparator}test:${containerRoot}test" `
+            --name "sdk-test-$appName" `
+            $fullSdkTag $sdkRunArg "${containerRoot}test${dirSeparator}create-run-publish-app.${testScriptSuffix}" "${containerRoot}${appName}" $sdkTag
+        if (-NOT $?) {
+            throw  "Testing $fullSdkTag failed"
+        }
 
-    Write-Host "----- Testing $baseTag-runtime-nanoserver -----"
-    docker run -t $optionalDockerRunArgs -v "$($appDir):c:\$appName" --name "runtime-test-$appName" --entrypoint dotnet "$baseTag-runtime-nanoserver" "C:\$appName\publish\$appName.dll"
-    if (-NOT $?) {
-        throw  "Testing $baseTag-runtime failed"
+        Write-Host "----- Testing $baseTag-runtime$tagSuffix with $sdkTag app -----"
+        docker run -t --rm `
+            -v "${appDir}:${containerRoot}${appName}" `
+            --name "runtime-test-$appName" `
+            --entrypoint dotnet `
+            "$baseTag-runtime$tagSuffix" `
+            "${containerRoot}${appName}${dirSeparator}publish${dirSeparator}framework-dependent${dirSeparator}${appName}.dll"
+        if (-NOT $?) {
+            throw  "Testing $baseTag-runtime failed"
+        }
+
+        if ($Platform -eq "linux") {
+            Write-Host "----- Testing $baseTag-runtime-deps with $sdkTag app -----"
+            docker run -t --rm `
+                -v "${appDir}:${containerRoot}${appName}" `
+                --name "runtime-deps-test-$appName" `
+                --entrypoint "${containerRoot}${appName}${dirSeparator}publish${dirSeparator}self-contained${dirSeparator}${appName}" `
+                "$baseTag-runtime-deps"
```

Comment:
What do you think about adding `$tagSuffix` here and in the `Write-Host` and `throw` statements?  This isn't necessary today, but if we add other platforms that have a runtime-deps image such as Alpine, it would be necessary, and I don't really see a significant downside to adding it.